lab_1/main.py

```diff
@@ -49,7 +72,27 @@ def get_top_n_words(freq_dict: dict, top_n: int) -> list:
     top_n = 1
     --> ['happy']
     """
-    pass
+    if isinstance(freq_dict, dict) and isinstance(top_n, int) and freq_dict and top_n >= 1:
+        items_list = list(freq_dict.items())
+
+        values_dict = {}
+        for pair in items_list:
+            if pair[1] in values_dict.keys():
+                values_dict[pair[1]].append(pair[0])
+            else:
+                values_dict[int(pair[1])] = [pair[0]]
+
+        top_n_list = list(values_dict.keys())
+        top_n_list.sort()
+
+        output_top = []
+        for ind in top_n_list:
+            output_top.extend(values_dict[ind][::-1])
+
+        output_top.reverse()
+        return output_top[:top_n]
+
+    return []
```

Comment:
nice, but difficult to understand. No need to change, just be ready to explain to me on October 3th